src/main/java/practice/CandidateValidator.java

```
@@ -1,5 +1,40 @@
 package practice;
 
-public class CandidateValidator {
-    //write your code here
+import java.util.function.Predicate;
+import model.Candidate;
+
+public class CandidateValidator implements Predicate<Candidate> {
+    private static final int MIN_AGE = 35;
+    private static final int MIN_PERIODS_IN_UKRAINE = 10;
+    private static final String NATIONALITY = "Ukrainian";
+    private static final String SEPARATOR_IN_PERIOD = "-";
+
+    /**
+     * Evaluates this predicate on the given argument.
+     *
+     * @param candidate the input argument
+     * @return {@code true} if the input argument matches the predicate,
+     * otherwise {@code false}
+     */
+    @Override
+    public boolean test(Candidate candidate) {
+        return candidate != null
+                && candidate.getAge() >= MIN_AGE
+                && candidate.isAllowedToVote()
+                && NATIONALITY.equals(candidate.getNationality())
+                && isValidPeriodsInUkraine(candidate.getPeriodsInUkr());
+    }
+
+    private boolean isValidPeriodsInUkraine(String periodInUkr) {
+        if (periodInUkr == null) {
+            return false;
+        }
+        String[] range = periodInUkr.split(SEPARATOR_IN_PERIOD);
+        try {
+            return Integer.parseInt(range[1])
+                    - Integer.parseInt(range[0]) + 1 >= MIN_PERIODS_IN_UKRAINE;
+        } catch (NumberFormatException e) {
+            return false;
+        }
```

catching RuntimeExceptions is not a good practice
can we throw a custom exception here? or just not catch it?

---

in this particular case we want the validator to do its job for any incoming data, if the period data cannot be recognized as numbers we simply return false. Throwing an exception doesn't make sense here.

---

Changed this method, simplified. If any exception occurs (when the input string does not match the format described in the task), false is returned.